### PR TITLE
Small fixes

### PR DESCRIPTION
**Pull request by shjortConcordium**
## Purpose

Small fixes to stuff I found during review.

## Changes

Added some tests for `parseKeyDerivationPath` in `util.c`, and mocks to allow that.
Added void parameter to some functions with no parameters.
Removed some unused functions.
Removed redundant string terminators in literals.
Changed/Added some comments.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.

### CODE REVIEW

File: src/signTransferWithSchedule.c [hunk 1]
```diff
@@ -199,13 +199,7 @@ void handleSignTransferWithSchedule(uint8_t *cdata, uint8_t p1, volatile unsigne
         ctx->state = TX_TRANSFER_WITH_SCHEDULE_INITIAL;
     }
 
-    if (memo_ctx->memoLength == 0 && ctx->state == TX_TRANSFER_WITH_SCHEDULE_MEMO) {
-        // Hash schedule length
-        cx_hash((cx_hash_t *) &tx_state->hash, 0, &ctx->remainingNumberOfScheduledAmounts, 1, NULL, 0);
```
Comment (orhoj) [inline src/signTransferWithSchedule.c:204]: Is the number of scheduled amounts ever hashed for the transaction with memo now?

File: src/signUpdateMintDistribution.c [hunk 2]
```diff
@@ -53,22 +53,22 @@ void handleSignUpdateMintDistribution(uint8_t *cdata, volatile unsigned int *fla
     uint8_t multiplication[6] = "*10^(-";
     memmove(ctx->mintRate + mintRateMantissaLength, multiplication, 6);
     int mintRateExponentLength = numberToText(ctx->mintRate + mintRateMantissaLength + 6, mintRateExponent);
-    uint8_t end_paranthesis[2] = ")\0";
+    uint8_t end_paranthesis[2] = ")";
```
Comment (orhoj) [inline src/signUpdateMintDistribution.c:56]: This should be camelCase.

File: src/util.c [hunk 3]
```diff
@@ -34,7 +34,7 @@ int parseKeyDerivationPath(uint8_t *cdata) {
     return 1 + (4 * keyPath->pathLength);
 }
 
-
+// TODO: This is only used in getPublicKey, so perhaps it should be moved there?
```
[the remark this answers is not in the document]
Comment (orhoj) [reply src/util.c:37]: I think it makes sense to move it.

File: unit_tests/tests/testTime.c [hunk 4]
```diff
@@ -63,6 +63,13 @@ static void test_secondsToTm_overflow() {
     assert_int_equal(result, -1);
 }
 
+static void test_memmove() {
```
Comment (orhoj) [inline unit_tests/tests/testTime.c:66]: I am not sure if this test is needed. But it definitely does not fit in this class.

Comment (shjortConcordium) [reply unit_tests/tests/testTime.c:66]: I removed it.

**Pull request merged**